src/aspire/volume/__init__.py

```diff
@@ -1,16 +1,164 @@
 import numpy as np
 
+from aspire.image import Image
 from aspire.nufft import Plan
 from aspire.utils import ensure
 from aspire.utils.coor_trans import grid_2d
 from aspire.utils.fft import centered_fft2, centered_ifft2
 from aspire.utils.matlab_compat import m_flatten, m_reshape
+from aspire.utils.preprocess import downsample
 
 
 class Volume:
     """
-    stack of volumes
+    Volume is an N x L x L x L array, along with associated utility methods.
     """
+    def __init__(self, data):
+        """
+        Create a volume initialized with data.
+
+        Volumes should be N x L x L x L,
+        or L x L x L which implies N=1.
+
+        :param data: Volume data
+
+        :return: A volume instance.
+        """
+
+        if data.ndim == 3:
+            data = data[np.newaxis, :, :, :]
+
+        ensure(data.ndim == 4,
+               'Volume data should be ndarray with shape NxLxLxL'
+               ' or LxLxL.')
+
+        ensure(data.shape[1] == data.shape[2] == data.shape[3],
+               'Only cubed ndarrays are supported.')
+
+        self.data = data
+        self.N = self.data.shape[0]
+        self.dtype = self.data.dtype
+        self.resolution = self.data.shape[1]
+        self.shape = self.data.shape
+        self.volume_shape = self.data.shape[1:]
+
+    def __getitem__(self, item):
+        # this is one reason why you might want Volume and VolumeStack classes...
+        #return Volume(self.data[item])
+        return self.data[item]
+
+    def __setitem__(self, key, value):
+        self.data[key] = value
+
+    def __len__(self):
+        return self.N
+
+    def __add__(self, other):
+        return Volume(self.data + other.data)
+
+    def __sub__(self, other):
+        return Volume(self.data - other.data)
+
+    def __mul__(self, other):
+        if isinstance(other, Volume):
+            res = Volume(self.data * other.data)
+        else:
+            res = Volume(self.data * other)
+
+        return res
+
+    # def __truediv__(self, other):
```

Comment:
Remove?

Reply:
sure, removed.  At some point there was some code using a `/` on a volume ndarray thing, but I got rid of it... wasn't sure if we wanted to keep that ability... seems no. (can use __mul__ for same thing anyway)